### PR TITLE
addresses #1141, unique-kmers.py being too noisy, bad naming paradigm for estimate functions

### PR DESCRIPTION
**Pull request by bocajnotnef**
see #1141 and #1106 

@ctb @luizirber CR/merge, please?

- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Is the Copyright year up to date?

### CODE REVIEW

File: oxli/functions.py [hunk 1]
```diff
@@ -12,40 +12,40 @@
...
-    H = int(M/intZ)
-    M = H*intZ
-    f2 = (1-math.exp(-N/float(H)))**intZ
+    N = math.log(2)*(M/float(K))
+    intN = int(N)
+    if intN == 0:
+        intN = 1
+    X = int(M/intN)
+    M = X*intN
+    f2 = (1-math.exp(-K/float(X)))**intN
```
Comment (luizirber) [inline oxli/functions.py]: `make pep8` doesn't complain about this, but `make autopep8` change these lines. Which one should we consider, @mr-c?

Comment (mr-c) [inline oxli/functions.py]: Probably time to re-pin the `pep8` version in `make install-dep`

Comment (luizirber) [inline oxli/functions.py]: 1.6.2 also don't catch it.

Comment (mr-c) [inline oxli/functions.py]: what are the changes that autopep8 makes?

Comment (luizirber) [inline oxli/functions.py]: Spaces around operators:
```diff
--- original/oxli/functions.py
+++ fixed/oxli/functions.py
@@ -17,13 +17,13 @@
     Utility function for estimating optimal counting table args where N is the
     number of unique kmer and M is the allotted amount of memory
     """
-    N = math.log(2)*(M/float(K))
+    N = math.log(2) * (M / float(K))
     intN = int(N)
     if intN == 0:
         intN = 1
-    X = int(M/intN)
-    M = X*intN
-    f2 = (1-math.exp(-K/float(X)))**intN
+    X = int(M / intN)
+    M = X * intN
+    f2 = (1 - math.exp(-K / float(X)))**intN
     res = namedtuple("result", ["num_htables", "htable_size", "mem_use",
                                 "fp_rate"])
     return res(intN, X, M, f2)
@@ -39,9 +39,9 @@
     if intN == 0:
         intN = 1
 
-    H1 = int(-K/(math.log(1-f**(1/float(intN)))))
+    H1 = int(-K / (math.log(1 - f**(1 / float(intN)))))
     M1 = H1 * intN
-    f1 = (1-math.exp(-K/float(H1)))**intN
+    f1 = (1 - math.exp(-K / float(H1)))**intN
 
     res = namedtuple("result", ["num_htables", "htable_size", "mem_use",
                                 "fp_rate"])
@@ -63,7 +63,8 @@
                     'expected_memory_usage')
 
     for fp_rate in range(1, 10):
-        Z, H, M, f = estimate_optimal_with_K_and_f(unique_kmers, fp_rate/10.0)
+        Z, H, M, f = estimate_optimal_with_K_and_f(
+            unique_kmers, fp_rate / 10.0)
         to_print.append('{:11.3f}\t{:19}\t{:17e}\t{:21e}'.format(f, Z, H, M))
 
     mem_list = [1, 5, 10, 20, 50, 100, 200, 300, 400, 500, 1000, 2000, 5000]
@@ -75,7 +76,7 @@
 
     for mem in mem_list:
         Z, H, M, f = estimate_optimal_with_K_and_M(unique_kmers,
-                                                   mem*1000000000)
+                                                   mem * 1000000000)
         to_print.append('{:21e}\t{:19}\t{:17e}\t{:11.3f}'.format(M, Z, H, f))
     return "\n".join(to_print)
```

Comment (mr-c) [inline oxli/functions.py]: Seems benign. Go for it.

File: oxli/functions.py [hunk 2]
```diff
@@ -12,40 +12,40 @@
 import khmer.utils
 
 
-def estimate_optimal_with_N_and_M(N, M):
+def estimate_optimal_with_K_and_M(K, M):
     """
     Utility function for estimating optimal counting table args where N is the
```
Comment (luizirber) [inline oxli/functions.py]: `K`, not `N`

File: oxli/functions.py [hunk 3]
```diff
@@ -12,40 +12,40 @@
...
     res = namedtuple("result", ["num_htables", "htable_size", "mem_use",
                                 "fp_rate"])
-    return res(intZ, H, M, f2)
+    return res(intN, X, M, f2)
 
 
-def estimate_optimal_with_N_and_f(N, f):
+def estimate_optimal_with_K_and_f(K, f):
     """
     Utility function for estimating optimal memory where N is the number of
```
Comment (luizirber) [inline oxli/functions.py]: `K`, not `N`

**Comment by bocajnotnef** [comment]
@luizirber Updated; linux build passing(ish)

File: oxli/functions.py [hunk 4]
```diff
@@ -12,11 +12,20 @@
...
 
 
-def estimate_optimal_with_N_and_M(N, M):
+def estimate_optimal_with_K_and_M(K, M):
     """
-    Utility function for estimating optimal counting table args where N is the
+    Utility function for estimating optimal counting table args where K is the
     number of unique kmer and M is the allotted amount of memory
     """
+<<<<<<< HEAD
```
Comment (luizirber) [inline oxli/functions.py]: =]

**Comment by bocajnotnef** [comment]
Merged #1161--Thanks @luizirber!

File: sandbox/unique-kmers.py [hunk 5]
```diff
@@ -108,9 +110,10 @@ def main():
     print('Total estimated number of unique {0}-mers: {1}'.format(
           args.ksize, cardinality),
           file=sys.stderr)
-    
-    to_print = output_gen(cardinality, args.error_rate)
-    print(to_print)
+
+    if(args.diagnostics):
```
Comment (luizirber) [inline sandbox/unique-kmers.py]: No need for parenthesis here

**Comment by luizirber** [comment]
LGTM (after https://github.com/dib-lab/khmer/pull/1145#discussion_r34217740)

@ctb what do you think?

File: ChangeLog [hunk 6]
```diff
@@ -1,3 +1,13 @@
+2015-07-08  Jacob Fenton  <bocajnotnef@gmail.com>
+
+   * oxli/functions.py: changed estimate functions to use correct letter
+   abbreviations
+   * sandbox/estimate_optimal_hash.py: changed to use renamed estimate 
+   functions
+   * sandbox/unique-kmers.py: changed to not output reccomended HT args by
```
Comment (luizirber) [inline ChangeLog]: typo

**Comment by bocajnotnef** [comment]
@luizirber @mr-c Merge/CR, please?

**Comment by bocajnotnef** [comment]
Thanks!

**Pull request merged**